template/app/src/main/java/com/xmartlabs/template/ui/common/BaseRecyclerViewAdapter.java

```diff
@@ -0,0 +1,204 @@
+package com.xmartlabs.template.ui.common;
+
+import android.support.annotation.CallSuper;
+import android.support.annotation.LayoutRes;
+import android.support.annotation.MainThread;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.xmartlabs.bigbang.core.helper.CollectionHelper;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+/**
+ * A Base RecyclerViewAdapter with already implemented functions such as
+ * Setting, removing, adding items, getting its count among others.
+ */
+@AllArgsConstructor
+@NoArgsConstructor
+public abstract class BaseRecyclerViewAdapter<T, VH extends RecyclerView.ViewHolder>
```

Comment:
yes, but it shouldn'tbe in the template! If you want move to ui, but not to the template